{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: qiime tools import [OPTIONS]\r\n",
      "\r\n",
      "  Import data to create a new QIIME 2 Artifact. See https://docs.qiime2.org/\r\n",
      "  for usage examples and details on the file types and associated semantic\r\n",
      "  types that can be imported.\r\n",
      "\r\n",
      "Options:\r\n",
      "  --type TEXT                The semantic type of the artifact that will be\r\n",
      "                             created upon importing. Use --show-importable-\r\n",
      "                             types to see what importable semantic types are\r\n",
      "                             available in the current deployment.  [required]\r\n",
      "  --input-path PATH          Path to file or directory that should be\r\n",
      "                             imported.  [required]\r\n",
      "  --output-path PATH         Path where output artifact should be written.\r\n",
      "                             [required]\r\n",
      "  --source-format TEXT       The format of the data to be imported. If not\r\n",
      "                             provided, data must be in the format expected by\r\n",
      "                             the semantic type provided via --type.\r\n",
      "  --show-importable-types    Show the semantic types that can be supplied to\r\n",
      "                             --type to import data into an artifact.\r\n",
      "  --show-importable-formats  Show formats that can be supplied to --source-\r\n",
      "                             format to import data into an artifact.\r\n",
      "  --help                     Show this message and exit.\r\n"
     ]
    }
   ],
   "source": [
    "!qiime tools import --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15S-NT-19O-1_GGATAAGA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-19O-1_GGATAAGA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-19O-1_GGATAAGA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-19O-1_GGATAAGA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-19O-2_TGCTGGAA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-19O-2_TGCTGGAA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-19O-2_TGCTGGAA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-19O-2_TGCTGGAA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-20O-A-1_GCAATTGG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-20O-A-1_GCAATTGG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-20O-A-1_GCAATTGG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-20O-A-1_GCAATTGG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-20O-A-2_CGCATGAG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-20O-A-2_CGCATGAG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-20O-A-2_CGCATGAG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-20O-A-2_CGCATGAG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-20O-B-1_TTATAGCA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-20O-B-1_TTATAGCA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-20O-B-1_TTATAGCA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-20O-B-1_TTATAGCA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-20O-B-2_CGCTATTA-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-20O-B-2_CGCTATTA-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-20O-B-2_CGCTATTA-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-20O-B-2_CGCTATTA-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-21M-1_TGGAATGC-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-NT-21M-1_TGGAATGC-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-21M-1_TGGAATGC-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-NT-21M-1_TGGAATGC-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-NT-21M-2_TGCCAGCG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-21M-2_TGCCAGCG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-21M-2_TGCCAGCG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-21M-2_TGCCAGCG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-21O-1_TACTCCTA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-21O-1_TACTCCTA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-21O-1_TACTCCTA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-21O-1_TACTCCTA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-21O-2_GTCCAGAA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-21O-2_GTCCAGAA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-21O-2_GTCCAGAA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-21O-2_GTCCAGAA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-22M-1_AGGCTGAA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-22M-1_AGGCTGAA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-22M-1_AGGCTGAA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-22M-1_AGGCTGAA-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-22M-2_CAATATAG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-22M-2_CAATATAG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-22M-2_CAATATAG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-22M-2_CAATATAG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-22O-1_TTAACGAG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-22O-1_TTAACGAG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-22O-1_TTAACGAG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-22O-1_TTAACGAG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-22O-2_CTCTCATG-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-22O-2_CTCTCATG-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-22O-2_CTCTCATG-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-22O-2_CTCTCATG-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-23M-1_GTTGCCAA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-23M-1_GTTGCCAA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-23M-1_GTTGCCAA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-23M-1_GTTGCCAA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-23M-2_GGATAAGA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-23M-2_GGATAAGA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-23M-2_GGATAAGA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-23M-2_GGATAAGA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-23O-1_TGCTGGAA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-23O-1_TGCTGGAA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-23O-1_TGCTGGAA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-23O-1_TGCTGGAA-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-23O-2_GCAATTGG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-23O-2_GCAATTGG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-23O-2_GCAATTGG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-23O-2_GCAATTGG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-24M-1_GTCTGCAG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-24M-1_GTCTGCAG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-24M-1_GTCTGCAG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-24M-1_GTCTGCAG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-24M-2_TTATAGCA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-24M-2_TTATAGCA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-24M-2_TTATAGCA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-24M-2_TTATAGCA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-24O-1_CCAACGTT-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-24O-1_CCAACGTT-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-24O-1_CCAACGTT-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-24O-1_CCAACGTT-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-24O-2_CGTTGAAG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-24O-2_CGTTGAAG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-24O-2_CGTTGAAG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-24O-2_CGTTGAAG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-25O-1_CGCTATTA-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-NT-25O-1_CGCTATTA-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-25O-1_CGCTATTA-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-NT-25O-1_CGCTATTA-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-NT-25O-2_TGGAATGC-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-25O-2_TGGAATGC-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-25O-2_TGGAATGC-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-25O-2_TGGAATGC-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-26O-1_TGCCAGCG-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-26O-1_TGCCAGCG-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-26O-1_TGCCAGCG-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-26O-1_TGCCAGCG-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-26O-2_ACTGCGCG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-26O-2_ACTGCGCG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-26O-2_ACTGCGCG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-26O-2_ACTGCGCG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-27M-1_TACTCCTA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-27M-1_TACTCCTA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-27M-1_TACTCCTA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-27M-1_TACTCCTA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-27M-2_GTCCAGAA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-27M-2_GTCCAGAA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-27M-2_GTCCAGAA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-27M-2_GTCCAGAA-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-27O-1_AGGCTGAA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-27O-1_AGGCTGAA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-27O-1_AGGCTGAA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-27O-1_AGGCTGAA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-27O-2_CAATATAG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-27O-2_CAATATAG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-27O-2_CAATATAG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-27O-2_CAATATAG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-28M-1_CTCTCATG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-28M-1_CTCTCATG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-28M-1_CTCTCATG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-28M-1_CTCTCATG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-28M-2_GTTGCCAA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-28M-2_GTTGCCAA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-28M-2_GTTGCCAA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-28M-2_GTTGCCAA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-28O-1_TGCTGGAA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-28O-1_TGCTGGAA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-28O-1_TGCTGGAA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-28O-1_TGCTGGAA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-28O-2_CGCATGAG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-28O-2_CGCATGAG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-28O-2_CGCATGAG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-28O-2_CGCATGAG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-29O-1_TTATAGCA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-29O-1_TTATAGCA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-29O-1_TTATAGCA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-29O-1_TTATAGCA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-29O-2_CGTTGAAG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-NT-29O-2_CGTTGAAG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-29O-2_CGTTGAAG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-NT-29O-2_CGTTGAAG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-NT-30M-1_CGCTATTA-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-30M-1_CGCTATTA-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-30M-1_CGCTATTA-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-30M-1_CGCTATTA-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-30M-2_TGGAATGC-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-30M-2_TGGAATGC-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-30M-2_TGGAATGC-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-30M-2_TGGAATGC-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-30O-1_TGCCAGCG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-30O-1_TGCCAGCG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-30O-1_TGCCAGCG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-30O-1_TGCCAGCG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-30O-2_ACTGCGCG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-30O-2_ACTGCGCG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-30O-2_ACTGCGCG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-30O-2_ACTGCGCG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-31M-1_TACTCCTA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-31M-1_TACTCCTA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-31M-1_TACTCCTA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-31M-1_TACTCCTA-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-31M-2_GTCCAGAA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-31M-2_GTCCAGAA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-31M-2_GTCCAGAA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-31M-2_GTCCAGAA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-31O-A-1_AGGCTGAA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-31O-A-1_AGGCTGAA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-31O-A-1_AGGCTGAA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-31O-A-1_AGGCTGAA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-31O-A-2_TTAACGAG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-31O-A-2_TTAACGAG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-31O-A-2_TTAACGAG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-31O-A-2_TTAACGAG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-31O-B-1_CTCTCATG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-31O-B-1_CTCTCATG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-31O-B-1_CTCTCATG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-31O-B-1_CTCTCATG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-31O-B-2_GTTGCCAA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-31O-B-2_GTTGCCAA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-31O-B-2_GTTGCCAA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-31O-B-2_GTTGCCAA-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-32O-1_GGATAAGA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-32O-1_GGATAAGA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-32O-1_GGATAAGA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-32O-1_GGATAAGA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-32O-2_TGCTGGAA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-32O-2_TGCTGGAA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-32O-2_TGCTGGAA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-32O-2_TGCTGGAA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-33M-1_GCAATTGG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-33M-1_GCAATTGG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-33M-1_GCAATTGG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-33M-1_GCAATTGG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-33M-2_CGCATGAG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-33M-2_CGCATGAG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-33M-2_CGCATGAG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-33M-2_CGCATGAG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-33O-1_GTCTGCAG-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-33O-1_GTCTGCAG-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-33O-1_GTCTGCAG-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-33O-1_GTCTGCAG-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-33O-2_CCAACGTT-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-33O-2_CCAACGTT-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-33O-2_CCAACGTT-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-33O-2_CCAACGTT-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-34M-1_AGTTCCGG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-NT-34M-1_AGTTCCGG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-34M-1_AGTTCCGG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-NT-34M-1_AGTTCCGG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-NT-34M-2_CGTTGAAG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-34M-2_CGTTGAAG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-34M-2_CGTTGAAG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-34M-2_CGTTGAAG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-34O-1_CGCTATTA-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-34O-1_CGCTATTA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-34O-1_CGCTATTA-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-34O-1_CGCTATTA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-34O-2_TGGAATGC-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-34O-2_TGGAATGC-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-34O-2_TGGAATGC-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-34O-2_TGGAATGC-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-35O-1_ACTGCGCG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-35O-1_ACTGCGCG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-35O-1_ACTGCGCG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-35O-1_ACTGCGCG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-35O-2_TACTCCTA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-35O-2_TACTCCTA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-35O-2_TACTCCTA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-35O-2_TACTCCTA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-36O-1_GTCCAGAA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-36O-1_GTCCAGAA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-36O-1_GTCCAGAA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-36O-1_GTCCAGAA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-36O-2_CAATATAG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-36O-2_CAATATAG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-36O-2_CAATATAG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-36O-2_CAATATAG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-38M-1_GTTGCCAA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-38M-1_GTTGCCAA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-38M-1_GTTGCCAA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-38M-1_GTTGCCAA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-38M-2_GGATAAGA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-38M-2_GGATAAGA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-38M-2_GGATAAGA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-38M-2_GGATAAGA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-38O-1_TTAACGAG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-38O-1_TTAACGAG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-38O-1_TTAACGAG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-38O-1_TTAACGAG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-38O-2_CTCTCATG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-38O-2_CTCTCATG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-38O-2_CTCTCATG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-38O-2_CTCTCATG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-39M-1_TTATAGCA-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-39M-1_TTATAGCA-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-39M-1_TTATAGCA-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-39M-1_TTATAGCA-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-39M-2_CCAACGTT-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-NT-39M-2_CCAACGTT-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-39M-2_CCAACGTT-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-NT-39M-2_CCAACGTT-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-NT-39O-1_AGTTCCGG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-39O-1_AGTTCCGG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-39O-1_AGTTCCGG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-39O-1_AGTTCCGG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-39O-2_CGTTGAAG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-39O-2_CGTTGAAG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-39O-2_CGTTGAAG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-39O-2_CGTTGAAG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-40M-1_CGCTATTA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-40M-1_CGCTATTA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-40M-1_CGCTATTA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-40M-1_CGCTATTA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-40M-2_TGGAATGC-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-40M-2_TGGAATGC-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-40M-2_TGGAATGC-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-40M-2_TGGAATGC-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-40O-1_TGCCAGCG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-40O-1_TGCCAGCG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-40O-1_TGCCAGCG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-40O-1_TGCCAGCG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-40O-2_ACTGCGCG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-40O-2_ACTGCGCG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-40O-2_ACTGCGCG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-40O-2_ACTGCGCG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-41M-1_TACTCCTA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-41M-1_TACTCCTA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-41M-1_TACTCCTA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-41M-1_TACTCCTA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-41M-2_AGGCTGAA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-41M-2_AGGCTGAA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-41M-2_AGGCTGAA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-41M-2_AGGCTGAA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-41O-1_CAATATAG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-41O-1_CAATATAG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-41O-1_CAATATAG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-41O-1_CAATATAG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-41O-2_TTAACGAG-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-41O-2_TTAACGAG-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-41O-2_TTAACGAG-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-41O-2_TTAACGAG-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-42M-1_CTCTCATG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-42M-1_CTCTCATG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-42M-1_CTCTCATG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-42M-1_CTCTCATG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-42M-2_GTTGCCAA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-42M-2_GTTGCCAA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-42M-2_GTTGCCAA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-42M-2_GTTGCCAA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-42O-1_TGCTGGAA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-42O-1_TGCTGGAA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-42O-1_TGCTGGAA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-42O-1_TGCTGGAA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-42O-2_GCAATTGG-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-42O-2_GCAATTGG-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-42O-2_GCAATTGG-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-42O-2_GCAATTGG-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-43M-1_ACTGCGCG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-43M-1_ACTGCGCG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-43M-1_ACTGCGCG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-43M-1_ACTGCGCG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-43M-2_GTCCAGAA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-43M-2_GTCCAGAA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-43M-2_GTCCAGAA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-43M-2_GTCCAGAA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-43O-1_CCAACGTT-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-43O-1_CCAACGTT-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-43O-1_CCAACGTT-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-43O-1_CCAACGTT-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-43O-2_AGTTCCGG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-43O-2_AGTTCCGG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-43O-2_AGTTCCGG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-43O-2_AGTTCCGG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-44M-1_CGTTGAAG-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-44M-1_CGTTGAAG-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-44M-1_CGTTGAAG-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-44M-1_CGTTGAAG-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-44M-2_CGCTATTA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-44M-2_CGCTATTA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-44M-2_CGCTATTA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-44M-2_CGCTATTA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-44O-1_TGGAATGC-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-44O-1_TGGAATGC-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-44O-1_TGGAATGC-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-44O-1_TGGAATGC-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-44O-2_TGCCAGCG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-44O-2_TGCCAGCG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-44O-2_TGCCAGCG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-44O-2_TGCCAGCG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-45O-1_AGGCTGAA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-45O-1_AGGCTGAA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-45O-1_AGGCTGAA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-45O-1_AGGCTGAA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-45O-2_CAATATAG-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-45O-2_CAATATAG-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-45O-2_CAATATAG-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-45O-2_CAATATAG-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-46M-1_CTCTCATG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-46M-1_CTCTCATG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-46M-1_CTCTCATG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-46M-1_CTCTCATG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-46M-2_GGATAAGA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-46M-2_GGATAAGA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-46M-2_GGATAAGA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-46M-2_GGATAAGA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-47M-1_CGCATGAG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-47M-1_CGCATGAG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-47M-1_CGCATGAG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-47M-1_CGCATGAG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-47M-2_TTATAGCA-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-47M-2_TTATAGCA-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-47M-2_TTATAGCA-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-47M-2_TTATAGCA-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-47O-1_CCAACGTT-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-47O-1_CCAACGTT-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-47O-1_CCAACGTT-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-47O-1_CCAACGTT-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-47O-2_AGTTCCGG-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-47O-2_AGTTCCGG-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-47O-2_AGTTCCGG-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-47O-2_AGTTCCGG-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-48O-1_CGTTGAAG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-48O-1_CGTTGAAG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-48O-1_CGTTGAAG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-48O-1_CGTTGAAG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-48O-2_CGCTATTA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-48O-2_CGCTATTA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-48O-2_CGCTATTA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-48O-2_CGCTATTA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-49M-1_TGGAATGC-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-49M-1_TGGAATGC-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-49M-1_TGGAATGC-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-49M-1_TGGAATGC-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-49M-2_TGCCAGCG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-49M-2_TGCCAGCG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-49M-2_TGCCAGCG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-49M-2_TGCCAGCG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-49O-1_TACTCCTA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-49O-1_TACTCCTA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-49O-1_TACTCCTA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-49O-1_TACTCCTA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-49O-2_GTCCAGAA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-49O-2_GTCCAGAA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-49O-2_GTCCAGAA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-49O-2_GTCCAGAA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-50O-1_AGGCTGAA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-50O-1_AGGCTGAA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-50O-1_AGGCTGAA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-50O-1_AGGCTGAA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-50O-2_TTAACGAG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-50O-2_TTAACGAG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-50O-2_TTAACGAG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-50O-2_TTAACGAG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-51O-1_GTTGCCAA-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-51O-1_GTTGCCAA-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-51O-1_GTTGCCAA-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-51O-1_GTTGCCAA-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-51O-2_GGATAAGA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-51O-2_GGATAAGA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-51O-2_GGATAAGA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-51O-2_GGATAAGA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-52O-1_TGCTGGAA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-52O-1_TGCTGGAA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-52O-1_TGCTGGAA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-52O-1_TGCTGGAA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-52O-2_GCAATTGG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-52O-2_GCAATTGG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-52O-2_GCAATTGG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-52O-2_GCAATTGG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U06O-1_CCAACGTT-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-U06O-1_CCAACGTT-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U06O-1_CCAACGTT-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-U06O-1_CCAACGTT-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U06O-2_AGTTCCGG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-U06O-2_AGTTCCGG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U06O-2_AGTTCCGG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-U06O-2_AGTTCCGG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U07M-1_CGCTATTA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-U07M-1_CGCTATTA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U07M-1_CGCTATTA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-U07M-1_CGCTATTA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U07M-2_TGGAATGC-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-U07M-2_TGGAATGC-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U07M-2_TGGAATGC-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-U07M-2_TGGAATGC-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U07O-1_TGCCAGCG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-NT-U07O-1_TGCCAGCG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U07O-1_TGCCAGCG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-NT-U07O-1_TGCCAGCG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U07O-2_ACTGCGCG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-U07O-2_ACTGCGCG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U07O-2_ACTGCGCG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-U07O-2_ACTGCGCG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U08M-1_TACTCCTA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-U08M-1_TACTCCTA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U08M-1_TACTCCTA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-U08M-1_TACTCCTA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U08M-2_AGGCTGAA-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-U08M-2_AGGCTGAA-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U08M-2_AGGCTGAA-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-U08M-2_AGGCTGAA-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U08O-1_CAATATAG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-NT-U08O-1_CAATATAG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U08O-1_CAATATAG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-NT-U08O-1_CAATATAG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U08O-2_TTAACGAG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-U08O-2_TTAACGAG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U08O-2_TTAACGAG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-U08O-2_TTAACGAG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U09M-1_CTCTCATG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-U09M-1_CTCTCATG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U09M-1_CTCTCATG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-U09M-1_CTCTCATG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U09M-2_GTTGCCAA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-NT-U09M-2_GTTGCCAA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U09M-2_GTTGCCAA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-NT-U09M-2_GTTGCCAA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U09O-1_TGCTGGAA-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-NT-U09O-1_TGCTGGAA-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U09O-1_TGCTGGAA-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-NT-U09O-1_TGCTGGAA-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U09O-2_GCAATTGG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-NT-U09O-2_GCAATTGG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U09O-2_GCAATTGG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-NT-U09O-2_GCAATTGG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U10M-1_CGCATGAG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-NT-U10M-1_CGCATGAG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U10M-1_CGCATGAG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-NT-U10M-1_CGCATGAG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U10M-2_GTCTGCAG-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-NT-U10M-2_GTCTGCAG-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U10M-2_GTCTGCAG-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-NT-U10M-2_GTCTGCAG-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U10O-1_TTATAGCA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-NT-U10O-1_TTATAGCA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U10O-1_TTATAGCA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-NT-U10O-1_TTATAGCA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U10O-2_CCAACGTT-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-NT-U10O-2_CCAACGTT-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U10O-2_CCAACGTT-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-NT-U10O-2_CCAACGTT-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U11O-1_AGTTCCGG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-NT-U11O-1_AGTTCCGG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U11O-1_AGTTCCGG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-NT-U11O-1_AGTTCCGG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U11O-2_CGTTGAAG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-NT-U11O-2_CGTTGAAG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U11O-2_CGTTGAAG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-NT-U11O-2_CGTTGAAG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U12O-1_CGCTATTA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-NT-U12O-1_CGCTATTA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U12O-1_CGCTATTA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-NT-U12O-1_CGCTATTA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-U12O-2_TGCCAGCG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-NT-U12O-2_TGCCAGCG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-U12O-2_TGCCAGCG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-NT-U12O-2_TGCCAGCG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-01M-1-10x_GGATAAGA-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-WB-01M-1-10x_GGATAAGA-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-01M-1-10x_GGATAAGA-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-WB-01M-1-10x_GGATAAGA-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-01M-2_CTCTCATG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-WB-01M-2_CTCTCATG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-01M-2_CTCTCATG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-WB-01M-2_CTCTCATG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-01O-1-10x_TGCTGGAA-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-WB-01O-1-10x_TGCTGGAA-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-01O-1-10x_TGCTGGAA-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-WB-01O-1-10x_TGCTGGAA-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-01O-2-10x_GCAATTGG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-01O-2-10x_GCAATTGG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-01O-2-10x_GCAATTGG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-01O-2-10x_GCAATTGG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-02O-1_GTTGCCAA-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-WB-02O-1_GTTGCCAA-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-02O-1_GTTGCCAA-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-WB-02O-1_GTTGCCAA-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-02O-2_GGATAAGA-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-02O-2_GGATAAGA-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-02O-2_GGATAAGA-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-02O-2_GGATAAGA-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-04M-1_TGCTGGAA-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-04M-1_TGCTGGAA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-04M-1_TGCTGGAA-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-04M-1_TGCTGGAA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-04M-2_GCAATTGG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-WB-04M-2_GCAATTGG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-04M-2_GCAATTGG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-WB-04M-2_GCAATTGG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-05M-1_CGCATGAG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-WB-05M-1_CGCATGAG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-05M-1_CGCATGAG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-WB-05M-1_CGCATGAG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-05M-2_GTCTGCAG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-WB-05M-2_GTCTGCAG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-05M-2_GTCTGCAG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-WB-05M-2_GTCTGCAG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-05O-1_TTATAGCA-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-WB-05O-1_TTATAGCA-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-05O-1_TTATAGCA-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-WB-05O-1_TTATAGCA-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-05O-2_CCAACGTT-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-WB-05O-2_CCAACGTT-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-05O-2_CCAACGTT-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-WB-05O-2_CCAACGTT-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-06M-1_CGTTGAAG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-WB-06M-1_CGTTGAAG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-06M-1_CGTTGAAG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-WB-06M-1_CGTTGAAG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-06M-2_CGCTATTA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-WB-06M-2_CGCTATTA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-06M-2_CGCTATTA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-WB-06M-2_CGCTATTA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-06O-1_TGGAATGC-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-WB-06O-1_TGGAATGC-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-06O-1_TGGAATGC-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-WB-06O-1_TGGAATGC-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-06O-2_TGCCAGCG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-WB-06O-2_TGCCAGCG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-06O-2_TGCCAGCG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-WB-06O-2_TGCCAGCG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-07M-1_ACTGCGCG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-WB-07M-1_ACTGCGCG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-07M-1_ACTGCGCG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-WB-07M-1_ACTGCGCG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-07M-2_TACTCCTA-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-07M-2_TACTCCTA-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-07M-2_TACTCCTA-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-07M-2_TACTCCTA-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-07O-1_GTCCAGAA-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-07O-1_GTCCAGAA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-07O-1_GTCCAGAA-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-07O-1_GTCCAGAA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-07O-2_AGGCTGAA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-07O-2_AGGCTGAA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-07O-2_AGGCTGAA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-07O-2_AGGCTGAA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-08M-1_CAATATAG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-WB-08M-1_CAATATAG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-08M-1_CAATATAG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-WB-08M-1_CAATATAG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-08M-2_TTAACGAG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-WB-08M-2_TTAACGAG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-08M-2_TTAACGAG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-WB-08M-2_TTAACGAG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-08O-1_CTCTCATG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-WB-08O-1_CTCTCATG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-08O-1_CTCTCATG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-WB-08O-1_CTCTCATG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-08O-2_GTTGCCAA-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-08O-2_GTTGCCAA-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-08O-2_GTTGCCAA-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-08O-2_GTTGCCAA-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-09M-1-R_GTCCAGAA-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-WB-09M-1-R_GTCCAGAA-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-09M-1-R_GTCCAGAA-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-WB-09M-1-R_GTCCAGAA-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-09M-2-R_AGGCTGAA-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-WB-09M-2-R_AGGCTGAA-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-09M-2-R_AGGCTGAA-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-WB-09M-2-R_AGGCTGAA-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-09O-1-R_CAATATAG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-09O-1-R_CAATATAG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-09O-1-R_CAATATAG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-09O-1-R_CAATATAG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-09O-2-R_TTAACGAG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-09O-2-R_TTAACGAG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-09O-2-R_TTAACGAG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-09O-2-R_TTAACGAG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-10M-1_TGCTGGAA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-10M-1_TGCTGGAA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-10M-1_TGCTGGAA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-10M-1_TGCTGGAA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-10M-2_GCAATTGG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-WB-10M-2_GCAATTGG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-10M-2_GCAATTGG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-WB-10M-2_GCAATTGG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-10O-1_CGCATGAG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-WB-10O-1_CGCATGAG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-10O-1_CGCATGAG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-WB-10O-1_CGCATGAG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-10O-2_GTCTGCAG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-WB-10O-2_GTCTGCAG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-10O-2_GTCTGCAG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-WB-10O-2_GTCTGCAG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-11M-1_TTATAGCA-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-WB-11M-1_TTATAGCA-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-11M-1_TTATAGCA-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-WB-11M-1_TTATAGCA-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-11M-2_AGTTCCGG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-WB-11M-2_AGTTCCGG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-11M-2_AGTTCCGG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-WB-11M-2_AGTTCCGG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-11O-1_CGTTGAAG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-WB-11O-1_CGTTGAAG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-11O-1_CGTTGAAG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-WB-11O-1_CGTTGAAG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-11O-2_CGCTATTA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-WB-11O-2_CGCTATTA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-11O-2_CGCTATTA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-WB-11O-2_CGCTATTA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-12M-1_TACTCCTA-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-12M-1_TACTCCTA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-12M-1_TACTCCTA-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-12M-1_TACTCCTA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-12M-2_GTCCAGAA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-12M-2_GTCCAGAA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-12M-2_GTCCAGAA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-12M-2_GTCCAGAA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-12O-1_AGGCTGAA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-WB-12O-1_AGGCTGAA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-12O-1_AGGCTGAA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-WB-12O-1_AGGCTGAA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-12O-2_CAATATAG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-WB-12O-2_CAATATAG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-12O-2_CAATATAG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-WB-12O-2_CAATATAG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-13O-1_TTAACGAG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-WB-13O-1_TTAACGAG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-13O-1_TTAACGAG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-WB-13O-1_TTAACGAG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-13O-2_CTCTCATG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-13O-2_CTCTCATG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-13O-2_CTCTCATG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-13O-2_CTCTCATG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-14M-1_GTTGCCAA-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-14M-1_GTTGCCAA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-14M-1_GTTGCCAA-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-14M-1_GTTGCCAA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-14M-2_GGATAAGA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-14M-2_GGATAAGA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-14M-2_GGATAAGA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-14M-2_GGATAAGA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-14O-1_TGCTGGAA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-WB-14O-1_TGCTGGAA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-14O-1_TGCTGGAA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-WB-14O-1_TGCTGGAA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-14O-2_GCAATTGG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-WB-14O-2_GCAATTGG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-14O-2_GCAATTGG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-WB-14O-2_GCAATTGG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-15M-1_CGCATGAG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-WB-15M-1_CGCATGAG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-15M-1_CGCATGAG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-WB-15M-1_CGCATGAG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-15M-2_GTCTGCAG-TGCCTACT_L001_R1_001.fastq\r\n",
      "15S-WB-15M-2_GTCTGCAG-TGCCTACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-15M-2_GTCTGCAG-TGCCTACT_L001_R2_001.fastq\r\n",
      "15S-WB-15M-2_GTCTGCAG-TGCCTACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-15O-1_AGTTCCGG-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-WB-15O-1_AGTTCCGG-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-15O-1_AGTTCCGG-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-WB-15O-1_AGTTCCGG-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-15O-2_CGTTGAAG-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-WB-15O-2_CGTTGAAG-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-15O-2_CGTTGAAG-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-WB-15O-2_CGTTGAAG-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-16M-1_TGGAATGC-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-WB-16M-1_TGGAATGC-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-16M-1_TGGAATGC-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-WB-16M-1_TGGAATGC-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-16M-2_TGCCAGCG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-16M-2_TGCCAGCG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-16M-2_TGCCAGCG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-16M-2_TGCCAGCG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-16O-1_ACTGCGCG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-16O-1_ACTGCGCG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-16O-1_ACTGCGCG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-16O-1_ACTGCGCG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-16O-2_TACTCCTA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-16O-2_TACTCCTA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-16O-2_TACTCCTA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-16O-2_TACTCCTA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-17M-1_GTCCAGAA-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-WB-17M-1_GTCCAGAA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-17M-1_GTCCAGAA-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-WB-17M-1_GTCCAGAA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-17M-2_AGGCTGAA-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-WB-17M-2_AGGCTGAA-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-17M-2_AGGCTGAA-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-WB-17M-2_AGGCTGAA-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-17O-1_CAATATAG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-WB-17O-1_CAATATAG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-17O-1_CAATATAG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-WB-17O-1_CAATATAG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-17O-2_TTAACGAG-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-WB-17O-2_TTAACGAG-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-17O-2_TTAACGAG-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-WB-17O-2_TTAACGAG-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-18M-1_CTCTCATG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-18M-1_CTCTCATG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-18M-1_CTCTCATG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-18M-1_CTCTCATG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-18M-2_GTTGCCAA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-18M-2_GTTGCCAA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-18M-2_GTTGCCAA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-18M-2_GTTGCCAA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U01M-1_GTCTGCAG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-U01M-1_GTCTGCAG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U01M-1_GTCTGCAG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-U01M-1_GTCTGCAG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U01M-2_TTATAGCA-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-U01M-2_TTATAGCA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U01M-2_TTATAGCA-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-U01M-2_TTATAGCA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U01O-1_CCAACGTT-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-U01O-1_CCAACGTT-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U01O-1_CCAACGTT-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-U01O-1_CCAACGTT-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U01O-2_AGTTCCGG-TTCTAAGT_L001_R1_001.fastq\r\n",
      "15S-WB-U01O-2_AGTTCCGG-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U01O-2_AGTTCCGG-TTCTAAGT_L001_R2_001.fastq\r\n",
      "15S-WB-U01O-2_AGTTCCGG-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U02O-1_CGTTGAAG-ACGCATTC_L001_R1_001.fastq\r\n",
      "15S-WB-U02O-1_CGTTGAAG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U02O-1_CGTTGAAG-ACGCATTC_L001_R2_001.fastq\r\n",
      "15S-WB-U02O-1_CGTTGAAG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U02O-2_CGCTATTA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "15S-WB-U02O-2_CGCTATTA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U02O-2_CGCTATTA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "15S-WB-U02O-2_CGCTATTA-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U03M-1_TGGAATGC-CTTACTCC_L001_R1_001.fastq\r\n",
      "15S-WB-U03M-1_TGGAATGC-CTTACTCC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U03M-1_TGGAATGC-CTTACTCC_L001_R2_001.fastq\r\n",
      "15S-WB-U03M-1_TGGAATGC-CTTACTCC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U03M-2_ACTGCGCG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-WB-U03M-2_ACTGCGCG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U03M-2_ACTGCGCG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-WB-U03M-2_ACTGCGCG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U03O-1_TACTCCTA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-WB-U03O-1_TACTCCTA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U03O-1_TACTCCTA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-WB-U03O-1_TACTCCTA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U03O-2_GTCCAGAA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-WB-U03O-2_GTCCAGAA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U03O-2_GTCCAGAA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-WB-U03O-2_GTCCAGAA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U04M-1_CAATATAG-AGAGACGT_L001_R1_001.fastq\r\n",
      "15S-WB-U04M-1_CAATATAG-AGAGACGT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U04M-1_CAATATAG-AGAGACGT_L001_R2_001.fastq\r\n",
      "15S-WB-U04M-1_CAATATAG-AGAGACGT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U04M-2_TTAACGAG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "15S-WB-U04M-2_TTAACGAG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U04M-2_TTAACGAG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "15S-WB-U04M-2_TTAACGAG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U04O-1_CTCTCATG-CTTAGCAG_L001_R1_001.fastq\r\n",
      "15S-WB-U04O-1_CTCTCATG-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U04O-1_CTCTCATG-CTTAGCAG_L001_R2_001.fastq\r\n",
      "15S-WB-U04O-1_CTCTCATG-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U04O-2_GTTGCCAA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "15S-WB-U04O-2_GTTGCCAA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U04O-2_GTTGCCAA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "15S-WB-U04O-2_GTTGCCAA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U05M-1_GGATAAGA-TTAGGCTC_L001_R1_001.fastq\r\n",
      "15S-WB-U05M-1_GGATAAGA-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U05M-1_GGATAAGA-TTAGGCTC_L001_R2_001.fastq\r\n",
      "15S-WB-U05M-1_GGATAAGA-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_R1_001.fastq\r\n",
      "15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_R2_001.fastq\r\n",
      "15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U05O-1_GTCTGCAG-CGTAGACT_L001_R1_001.fastq\r\n",
      "15S-WB-U05O-1_GTCTGCAG-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U05O-1_GTCTGCAG-CGTAGACT_L001_R2_001.fastq\r\n",
      "15S-WB-U05O-1_GTCTGCAG-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U05O-2_TTATAGCA-GGATAAGG_L001_R1_001.fastq\r\n",
      "15S-WB-U05O-2_TTATAGCA-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U05O-2_TTATAGCA-GGATAAGG_L001_R2_001.fastq\r\n",
      "15S-WB-U05O-2_TTATAGCA-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "Blank-10_TGCCAGCG-ACGCATTC_L001_R1_001.fastq\r\n",
      "Blank-10_TGCCAGCG-ACGCATTC_L001_R1_001.fastq.gz\r\n",
      "Blank-10_TGCCAGCG-ACGCATTC_L001_R2_001.fastq\r\n",
      "Blank-10_TGCCAGCG-ACGCATTC_L001_R2_001.fastq.gz\r\n",
      "Blank-11_TTATAGCA-CCAAGCTA_L001_R1_001.fastq\r\n",
      "Blank-11_TTATAGCA-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "Blank-11_TTATAGCA-CCAAGCTA_L001_R2_001.fastq\r\n",
      "Blank-11_TTATAGCA-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "Blank-13_TGCTGGAA-TCTTGAAT_L001_R1_001.fastq\r\n",
      "Blank-13_TGCTGGAA-TCTTGAAT_L001_R1_001.fastq.gz\r\n",
      "Blank-13_TGCTGGAA-TCTTGAAT_L001_R2_001.fastq\r\n",
      "Blank-13_TGCTGGAA-TCTTGAAT_L001_R2_001.fastq.gz\r\n",
      "Blank-16_CGCATGAG-CCAAGCTA_L001_R1_001.fastq\r\n",
      "Blank-16_CGCATGAG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "Blank-16_CGCATGAG-CCAAGCTA_L001_R2_001.fastq\r\n",
      "Blank-16_CGCATGAG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "Blank-17_CGTTGAAG-ACGCGCGT_L001_R1_001.fastq\r\n",
      "Blank-17_CGTTGAAG-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "Blank-17_CGTTGAAG-ACGCGCGT_L001_R2_001.fastq\r\n",
      "Blank-17_CGTTGAAG-ACGCGCGT_L001_R2_001.fastq.gz\r\n",
      "Blank-1_GGATAAGA-CGTAGACT_L001_R1_001.fastq\r\n",
      "Blank-1_GGATAAGA-CGTAGACT_L001_R1_001.fastq.gz\r\n",
      "Blank-1_GGATAAGA-CGTAGACT_L001_R2_001.fastq\r\n",
      "Blank-1_GGATAAGA-CGTAGACT_L001_R2_001.fastq.gz\r\n",
      "Blank-22_ACTGCGCG-TTAGGCTC_L001_R1_001.fastq\r\n",
      "Blank-22_ACTGCGCG-TTAGGCTC_L001_R1_001.fastq.gz\r\n",
      "Blank-22_ACTGCGCG-TTAGGCTC_L001_R2_001.fastq\r\n",
      "Blank-22_ACTGCGCG-TTAGGCTC_L001_R2_001.fastq.gz\r\n",
      "Blank-4_CCAACGTT-CTTAGCAG_L001_R1_001.fastq\r\n",
      "Blank-4_CCAACGTT-CTTAGCAG_L001_R1_001.fastq.gz\r\n",
      "Blank-4_CCAACGTT-CTTAGCAG_L001_R2_001.fastq\r\n",
      "Blank-4_CCAACGTT-CTTAGCAG_L001_R2_001.fastq.gz\r\n",
      "Blank-6_ACTGCGCG-GGATAAGG_L001_R1_001.fastq\r\n",
      "Blank-6_ACTGCGCG-GGATAAGG_L001_R1_001.fastq.gz\r\n",
      "Blank-6_ACTGCGCG-GGATAAGG_L001_R2_001.fastq\r\n",
      "Blank-6_ACTGCGCG-GGATAAGG_L001_R2_001.fastq.gz\r\n",
      "Blank-8_GGATAAGA-ACGCGCGT_L001_R1_001.fastq\r\n",
      "Blank-8_GGATAAGA-ACGCGCGT_L001_R1_001.fastq.gz\r\n",
      "Blank-8_GGATAAGA-ACGCGCGT_L001_R2_001.fastq\r\n",
      "Blank-8_GGATAAGA-ACGCGCGT_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls ../data/Seq_data/ITS2/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!mkdir ../data/Seq_data/ITS2mini"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cp ../data/Seq_data/ITS2/15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_*_001.fastq.gz ../data/Seq_data/ITS2mini"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15S-NT-19O-1_GGATAAGA-TTCTAAGT_L001_R1_001.fastq.gz\r\n",
      "15S-NT-19O-1_GGATAAGA-TTCTAAGT_L001_R2_001.fastq.gz\r\n",
      "15S-NT-21M-1_TGGAATGC-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-NT-21M-1_TGGAATGC-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-08O-1_CTCTCATG-CCAAGCTA_L001_R1_001.fastq.gz\r\n",
      "15S-WB-08O-1_CTCTCATG-CCAAGCTA_L001_R2_001.fastq.gz\r\n",
      "15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_R1_001.fastq.gz\r\n",
      "15S-WB-U05M-2_CGCATGAG-GGCTCGAT_L001_R2_001.fastq.gz\r\n"
     ]
    }
   ],
   "source": [
    "!ls ../data/Seq_data/ITS2mini/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!qiime tools import  --type 'SampleData[PairedEndSequencesWithQuality]' --input-path ../data/Seq_data/ITS2mini --source-format CasavaOneEightSingleLanePerSampleDirFmt --output-path demux.mini.qza\n",
    "# Here we are importing our data\n",
    "# It's in a different format than the data from the tutorial\n",
    "# We received our files from the sequencing centre already demultiplexed -\n",
    "# that is, there is a separate pair of .fastq.gz files (forward and reverse read) for each of our samples.\n",
    "\n",
    "# Note they are already demultiplexed. So I just called them demux.qza"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: demux.qzv\u001b[0m\r\n"
     ]
    }
   ],
   "source": [
    "!qiime demux summarize --i-data demux.qza --o-visualization demux.qzv --verbose\n",
    "\n",
    "# Old command: !qiime dada2 plot-qualities --verbose --i-demultiplexed-seqs demux.qza --p-n 4 --o-visualization demux-qualities.qzv\n",
    "# This command will create plots of the quality scores for our sequences (4 randomly chosen ones)\n",
    "# It will be output as demux-qual-plots.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting.\n",
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting."
     ]
    }
   ],
   "source": [
    "!qiime tools view demux.qzv\n",
    "# Let's take a look at the read quality plots.\n",
    "# How long are the reads?\n",
    "# Where along the read do sequences get bad?\n",
    "# Do the forward or reverse reads tend to be better quality?\n",
    "\n",
    "# A few thoughts:\n",
    "# Part of the reason the average q score is lower than our 16S data may be beacuse of the 300bp Illumina issues (vs 250)\n",
    "# I.e., it might not be our primers' fault.\n",
    "# Q scores aren't amazing, aren't atrocious.\n",
    "# Worth assembling...\n",
    "\n",
    "\n",
    "# The data look basically fine - the first 5 reads are lower quality (likely bp homeogeneity)\n",
    "# - can trim these in next step.\n",
    "\n",
    "# Remember you have to press the square (\"STOP\") button to stop this cell running the visualization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: qiime dada2 denoise-paired [OPTIONS]\r\n",
      "\r\n",
      "  This method denoises paired-end sequences, dereplicates them, and filters\r\n",
      "  chimeras.\r\n",
      "\r\n",
      "Options:\r\n",
      "  --i-demultiplexed-seqs PATH     Artifact:\r\n",
      "                                  SampleData[PairedEndSequencesWithQuality]\r\n",
      "                                  [required]\r\n",
      "                                  The paired-end demultiplexed\r\n",
      "                                  sequences to be denoised.\r\n",
      "  --p-trunc-len-f INTEGER         [required]\r\n",
      "                                  Position at which forward read\r\n",
      "                                  sequences should be truncated due to\r\n",
      "                                  decrease in quality. This truncates the 3'\r\n",
      "                                  end of the of the input sequences, which\r\n",
      "                                  will be the bases that were sequenced in the\r\n",
      "                                  last cycles. Reads that are shorter than\r\n",
      "                                  this value will be discarded. After this\r\n",
      "                                  parameter is applied there must still be at\r\n",
      "                                  least a 20 nucleotide overlap between the\r\n",
      "                                  forward and reverse reads. If 0 is provided,\r\n",
      "                                  no truncation or length filtering will be\r\n",
      "                                  performed\r\n",
      "  --p-trunc-len-r INTEGER         [required]\r\n",
      "                                  Position at which reverse read\r\n",
      "                                  sequences should be truncated due to\r\n",
      "                                  decrease in quality. This truncates the 3'\r\n",
      "                                  end of the of the input sequences, which\r\n",
      "                                  will be the bases that were sequenced in the\r\n",
      "                                  last cycles. Reads that are shorter than\r\n",
      "                                  this value will be discarded. After this\r\n",
      "                                  parameter is applied there must still be at\r\n",
      "                                  least a 20 nucleotide overlap between the\r\n",
      "                                  forward and reverse reads. If 0 is provided,\r\n",
      "                                  no truncation or length filtering will be\r\n",
      "                                  performed\r\n",
      "  --p-trim-left-f INTEGER         [default: 0]\r\n",
      "                                  Position at which forward read\r\n",
      "                                  sequences should be trimmed due to low\r\n",
      "                                  quality. This trims the 5' end of the input\r\n",
      "                                  sequences, which will be the bases that were\r\n",
      "                                  sequenced in the first cycles.\r\n",
      "  --p-trim-left-r INTEGER         [default: 0]\r\n",
      "                                  Position at which reverse read\r\n",
      "                                  sequences should be trimmed due to low\r\n",
      "                                  quality. This trims the 5' end of the input\r\n",
      "                                  sequences, which will be the bases that were\r\n",
      "                                  sequenced in the first cycles.\r\n",
      "  --p-max-ee FLOAT                [default: 2.0]\r\n",
      "                                  Reads with number of expected\r\n",
      "                                  errors higher than this value will be\r\n",
      "                                  discarded.\r\n",
      "  --p-trunc-q INTEGER             [default: 2]\r\n",
      "                                  Reads are truncated at the\r\n",
      "                                  first instance of a quality score less than\r\n",
      "                                  or equal to this value. If the resulting\r\n",
      "                                  read is then shorter than `trunc_len_f` or\r\n",
      "                                  `trunc_len_r` (depending on the direction of\r\n",
      "                                  the read) it is discarded.\r\n",
      "  --p-chimera-method [consensus|pooled|none]\r\n",
      "                                  [default: consensus]\r\n",
      "                                  The method used to\r\n",
      "                                  remove chimeras. \"none\": No chimera removal\r\n",
      "                                  is performed. \"pooled\": All reads are pooled\r\n",
      "                                  prior to chimera detection. \"consensus\":\r\n",
      "                                  Chimeras are detected in samples\r\n",
      "                                  individually, and sequences found chimeric\r\n",
      "                                  in a sufficient fraction of samples are\r\n",
      "                                  removed.\r\n",
      "  --p-min-fold-parent-over-abundance FLOAT\r\n",
      "                                  [default: 1.0]\r\n",
      "                                  The minimum abundance of\r\n",
      "                                  potential parents of a sequence being tested\r\n",
      "                                  as chimeric, expressed as a fold-change\r\n",
      "                                  versus the abundance of the sequence being\r\n",
      "                                  tested. Values should be greater than or\r\n",
      "                                  equal to 1 (i.e. parents should be more\r\n",
      "                                  abundant than the sequence being tested).\r\n",
      "                                  This parameter has no effect if\r\n",
      "                                  chimera_method is \"none\".\r\n",
      "  --p-n-threads INTEGER           [default: 1]\r\n",
      "                                  The number of threads to use\r\n",
      "                                  for multithreaded processing. If 0 is\r\n",
      "                                  provided, all available cores will be used.\r\n",
      "  --p-n-reads-learn INTEGER       [default: 1000000]\r\n",
      "                                  The number of reads to\r\n",
      "                                  use when training the error model. Smaller\r\n",
      "                                  numbers will result in a shorter run time\r\n",
      "                                  but a less reliable error model.\r\n",
      "  --p-hashed-feature-ids / --p-no-hashed-feature-ids\r\n",
      "                                  [default: True]\r\n",
      "                                  If true, the feature ids in\r\n",
      "                                  the resulting table will be presented as\r\n",
      "                                  hashes of the sequences defining each\r\n",
      "                                  feature. The hash will always be the same\r\n",
      "                                  for the same sequence so this allows feature\r\n",
      "                                  tables to be merged across runs of this\r\n",
      "                                  method. You should only merge tables if the\r\n",
      "                                  exact same parameters are used for each run.\r\n",
      "  --o-table PATH                  Artifact: FeatureTable[Frequency]  [required\r\n",
      "                                  if not passing --output-dir]\r\n",
      "                                  The resulting\r\n",
      "                                  feature table.\r\n",
      "  --o-representative-sequences PATH\r\n",
      "                                  Artifact: FeatureData[Sequence]  [required\r\n",
      "                                  if not passing --output-dir]\r\n",
      "                                  The resulting\r\n",
      "                                  feature sequences. Each feature in the\r\n",
      "                                  feature table will be represented by exactly\r\n",
      "                                  one sequence, and these sequences will be\r\n",
      "                                  the joined paired-end sequences.\r\n",
      "  --output-dir DIRECTORY          Output unspecified results to a directory\r\n",
      "  --cmd-config PATH               Use config file for command options\r\n",
      "  --verbose                       Display verbose output to stdout and/or\r\n",
      "                                  stderr during execution of this action.\r\n",
      "                                  [default: False]\r\n",
      "  --quiet                         Silence output if execution is successful\r\n",
      "                                  (silence is golden).  [default: False]\r\n",
      "  --help                          Show this message and exit.\r\n"
     ]
    }
   ],
   "source": [
    "!qiime dada2 denoise-paired --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running external command line application(s). This may print messages to stdout and/or stderr.\n",
      "The command(s) being run are below. These commands cannot be manually re-run as they will depend on temporary files that no longer exist.\n",
      "\n",
      "Command: run_dada_paired.R /var/folders/d2/qqsv2qxd5fjf4k455pzytwgh0000gn/T/tmpqmyelgqp/forward /var/folders/d2/qqsv2qxd5fjf4k455pzytwgh0000gn/T/tmpqmyelgqp/reverse /var/folders/d2/qqsv2qxd5fjf4k455pzytwgh0000gn/T/tmpqmyelgqp/output.tsv.biom /var/folders/d2/qqsv2qxd5fjf4k455pzytwgh0000gn/T/tmpqmyelgqp/filt_f /var/folders/d2/qqsv2qxd5fjf4k455pzytwgh0000gn/T/tmpqmyelgqp/filt_r 0 0 0 0 5.0 2 consensus 1.0 0 3000000\n",
      "\n",
      "R version 3.3.1 (2016-06-21) \n",
      "Loading required package: Rcpp\n",
      "Warning messages:\n",
      "1: multiple methods tables found for arbind \n",
      "2: multiple methods tables found for acbind \n",
      "3: replacing previous import IRanges::arbind by SummarizedExperiment::arbind when loading GenomicAlignments \n",
      "4: replacing previous import IRanges::acbind by SummarizedExperiment::acbind when loading GenomicAlignments \n",
      "5: multiple methods tables found for left \n",
      "6: multiple methods tables found for right \n",
      "DADA2 R package version: 1.4.0 \n",
      "1) Filtering ..........................................................................................................................................................................................................................\n",
      "2) Learning Error Rates\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "Not all sequences were the same length.\n",
      "2a) Forward Reads\n",
      "Initializing error rates to maximum possible estimate.\n",
      "Sample 1 - 23019 reads in 19469 unique sequences.\n",
      "Sample 2 - 8541 reads in 7817 unique sequences.\n",
      "Sample 3 - 17795 reads in 16316 unique sequences.\n",
      "Sample 4 - 10717 reads in 9738 unique sequences.\n",
      "Sample 5 - 18306 reads in 16279 unique sequences.\n",
      "Sample 6 - 11676 reads in 10684 unique sequences.\n",
      "Sample 7 - 9082 reads in 8349 unique sequences.\n",
      "Sample 8 - 9944 reads in 9213 unique sequences.\n",
      "Sample 9 - 18467 reads in 12807 unique sequences.\n",
      "Sample 10 - 11020 reads in 8380 unique sequences.\n",
      "Sample 11 - 162 reads in 157 unique sequences.\n",
      "Sample 12 - 9990 reads in 9268 unique sequences.\n",
      "Sample 13 - 12309 reads in 11326 unique sequences.\n",
      "Sample 14 - 17294 reads in 15247 unique sequences.\n",
      "Sample 15 - 15253 reads in 13477 unique sequences.\n",
      "Sample 16 - 13013 reads in 11375 unique sequences.\n",
      "Sample 17 - 16632 reads in 14853 unique sequences.\n",
      "Sample 18 - 12699 reads in 11388 unique sequences.\n",
      "Sample 19 - 14950 reads in 13108 unique sequences.\n",
      "Sample 20 - 17943 reads in 15474 unique sequences.\n",
      "Sample 21 - 15274 reads in 13394 unique sequences.\n",
      "Sample 22 - 8519 reads in 7459 unique sequences.\n",
      "Sample 23 - 20508 reads in 17623 unique sequences.\n",
      "Sample 24 - 14753 reads in 12876 unique sequences.\n",
      "Sample 25 - 14557 reads in 12759 unique sequences.\n",
      "Sample 26 - 12832 reads in 9293 unique sequences.\n",
      "Sample 27 - 6172 reads in 4813 unique sequences.\n",
      "Sample 28 - 10076 reads in 8812 unique sequences.\n",
      "Sample 29 - 15731 reads in 13986 unique sequences.\n",
      "Sample 30 - 14988 reads in 12860 unique sequences.\n",
      "Sample 31 - 17632 reads in 14261 unique sequences.\n",
      "Sample 32 - 9583 reads in 8752 unique sequences.\n",
      "Sample 33 - 33268 reads in 28831 unique sequences.\n",
      "Sample 34 - 16062 reads in 14324 unique sequences.\n",
      "Sample 35 - 2 reads in 2 unique sequences.\n",
      "Sample 36 - 17794 reads in 15620 unique sequences.\n",
      "Sample 37 - 23211 reads in 19597 unique sequences.\n",
      "Sample 38 - 14487 reads in 12793 unique sequences.\n",
      "Sample 39 - 18197 reads in 15758 unique sequences.\n",
      "Sample 40 - 11899 reads in 10754 unique sequences.\n",
      "Sample 41 - 10514 reads in 9525 unique sequences.\n",
      "Sample 42 - 30090 reads in 26602 unique sequences.\n",
      "Sample 43 - 25908 reads in 22958 unique sequences.\n",
      "Sample 44 - 10997 reads in 9769 unique sequences.\n",
      "Sample 45 - 8958 reads in 7981 unique sequences.\n",
      "Sample 46 - 9653 reads in 7184 unique sequences.\n",
      "Sample 47 - 17169 reads in 12232 unique sequences.\n",
      "Sample 48 - 19525 reads in 17717 unique sequences.\n",
      "Sample 49 - 10079 reads in 9262 unique sequences.\n",
      "Sample 50 - 15964 reads in 13551 unique sequences.\n",
      "Sample 51 - 12976 reads in 10873 unique sequences.\n",
      "Sample 52 - 21186 reads in 18448 unique sequences.\n",
      "Sample 53 - 19489 reads in 17502 unique sequences.\n",
      "Sample 54 - 20982 reads in 18812 unique sequences.\n",
      "Sample 55 - 15116 reads in 13476 unique sequences.\n",
      "Sample 56 - 15638 reads in 13337 unique sequences.\n",
      "Sample 57 - 8881 reads in 7450 unique sequences.\n",
      "Sample 58 - 16211 reads in 13637 unique sequences.\n",
      "Sample 59 - 7 reads in 7 unique sequences.\n",
      "Sample 60 - 3634 reads in 3307 unique sequences.\n",
      "Sample 61 - 13650 reads in 12051 unique sequences.\n",
      "Sample 62 - 16223 reads in 12226 unique sequences.\n",
      "Sample 63 - 19122 reads in 16265 unique sequences.\n",
      "Sample 64 - 14891 reads in 12811 unique sequences.\n",
      "Sample 65 - 18187 reads in 13617 unique sequences.\n",
      "Sample 66 - 14281 reads in 12984 unique sequences.\n",
      "Sample 67 - 18784 reads in 16628 unique sequences.\n",
      "Sample 68 - 9472 reads in 8419 unique sequences.\n",
      "Sample 69 - 9864 reads in 8917 unique sequences.\n",
      "Sample 70 - 27853 reads in 24026 unique sequences.\n",
      "Sample 71 - 5381 reads in 5032 unique sequences.\n",
      "Sample 72 - 9324 reads in 8526 unique sequences.\n",
      "Sample 73 - 6624 reads in 6070 unique sequences.\n",
      "Sample 74 - 9045 reads in 8375 unique sequences.\n",
      "Sample 75 - 14406 reads in 12888 unique sequences.\n",
      "Sample 76 - 11569 reads in 10266 unique sequences.\n",
      "Sample 77 - 9786 reads in 8974 unique sequences.\n",
      "Sample 78 - 27061 reads in 24576 unique sequences.\n",
      "Sample 79 - 15770 reads in 14279 unique sequences.\n",
      "Sample 80 - 8457 reads in 7820 unique sequences.\n",
      "Sample 81 - 11 reads in 11 unique sequences.\n",
      "Sample 82 - 13 reads in 13 unique sequences.\n",
      "Sample 83 - 8926 reads in 8155 unique sequences.\n",
      "Sample 84 - 6880 reads in 6359 unique sequences.\n",
      "Sample 85 - 7 reads in 7 unique sequences.\n",
      "Sample 86 - 4 reads in 4 unique sequences.\n",
      "Sample 87 - 2 reads in 2 unique sequences.\n",
      "Sample 88 - 5 reads in 5 unique sequences.\n",
      "Sample 89 - 15939 reads in 14708 unique sequences.\n",
      "Sample 90 - 9472 reads in 8292 unique sequences.\n",
      "Sample 91 - 10720 reads in 9334 unique sequences.\n",
      "Sample 92 - 18560 reads in 14513 unique sequences.\n",
      "Sample 93 - 10087 reads in 8977 unique sequences.\n",
      "Sample 94 - 10315 reads in 9106 unique sequences.\n",
      "Sample 95 - 7 reads in 7 unique sequences.\n",
      "Sample 96 - 7283 reads in 6631 unique sequences.\n",
      "Sample 97 - 12678 reads in 10919 unique sequences.\n",
      "Sample 98 - 21797 reads in 19552 unique sequences.\n",
      "Sample 99 - 9818 reads in 9069 unique sequences.\n",
      "Sample 100 - 18792 reads in 17007 unique sequences.\n",
      "Sample 101 - 40141 reads in 33164 unique sequences.\n",
      "Sample 102 - 36355 reads in 32908 unique sequences.\n",
      "Sample 103 - 613 reads in 603 unique sequences.\n",
      "Sample 104 - 6690 reads in 6174 unique sequences.\n",
      "Sample 105 - 9309 reads in 8433 unique sequences.\n",
      "Sample 106 - 1305 reads in 1022 unique sequences.\n",
      "Sample 107 - 945 reads in 757 unique sequences.\n",
      "Sample 108 - 6021 reads in 4766 unique sequences.\n",
      "Sample 109 - 6794 reads in 6024 unique sequences.\n",
      "Sample 110 - 5219 reads in 4906 unique sequences.\n",
      "Sample 111 - 12329 reads in 11159 unique sequences.\n",
      "Sample 112 - 17092 reads in 15416 unique sequences.\n",
      "Sample 113 - 68 reads in 65 unique sequences.\n",
      "Sample 114 - 16131 reads in 14264 unique sequences.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample 115 - 12525 reads in 11018 unique sequences.\n",
      "Sample 116 - 12921 reads in 11646 unique sequences.\n",
      "Sample 117 - 12197 reads in 10995 unique sequences.\n",
      "Sample 118 - 9762 reads in 8518 unique sequences.\n",
      "Sample 119 - 12402 reads in 10848 unique sequences.\n",
      "Sample 120 - 11245 reads in 10022 unique sequences.\n",
      "Sample 121 - 4807 reads in 4435 unique sequences.\n",
      "Sample 122 - 642 reads in 609 unique sequences.\n",
      "Sample 123 - 15606 reads in 13168 unique sequences.\n",
      "Sample 124 - 15298 reads in 12745 unique sequences.\n",
      "Sample 125 - 12000 reads in 10049 unique sequences.\n",
      "Sample 126 - 9152 reads in 7637 unique sequences.\n",
      "Sample 127 - 8130 reads in 7000 unique sequences.\n",
      "Sample 128 - 11297 reads in 8995 unique sequences.\n",
      "Sample 129 - 14584 reads in 11688 unique sequences.\n",
      "Sample 130 - 9790 reads in 9084 unique sequences.\n",
      "Sample 131 - 14328 reads in 12900 unique sequences.\n",
      "Sample 132 - 27570 reads in 22224 unique sequences.\n",
      "Sample 133 - 13890 reads in 12361 unique sequences.\n",
      "Sample 134 - 21011 reads in 17943 unique sequences.\n",
      "Sample 135 - 25225 reads in 20754 unique sequences.\n",
      "Sample 136 - 26349 reads in 22749 unique sequences.\n",
      "Sample 137 - 30769 reads in 26472 unique sequences.\n",
      "Sample 138 - 23586 reads in 20758 unique sequences.\n",
      "Sample 139 - 14470 reads in 13204 unique sequences.\n",
      "Sample 140 - 16157 reads in 14538 unique sequences.\n",
      "Sample 141 - 46411 reads in 40458 unique sequences.\n",
      "Sample 142 - 26002 reads in 20159 unique sequences.\n",
      "Sample 143 - 15312 reads in 12199 unique sequences.\n",
      "Sample 144 - 23646 reads in 20486 unique sequences.\n",
      "Sample 145 - 14398 reads in 12615 unique sequences.\n",
      "Sample 146 - 14621 reads in 10502 unique sequences.\n",
      "Sample 147 - 20055 reads in 14430 unique sequences.\n",
      "Sample 148 - 15231 reads in 13509 unique sequences.\n",
      "Sample 149 - 13383 reads in 11916 unique sequences.\n",
      "Sample 150 - 16846 reads in 14751 unique sequences.\n",
      "Sample 151 - 12330 reads in 10687 unique sequences.\n",
      "Sample 152 - 11638 reads in 10628 unique sequences.\n",
      "Sample 153 - 13862 reads in 12669 unique sequences.\n",
      "Sample 154 - 14304 reads in 11899 unique sequences.\n",
      "Sample 155 - 16302 reads in 13084 unique sequences.\n",
      "Sample 156 - 10116 reads in 8735 unique sequences.\n",
      "Sample 157 - 12 reads in 11 unique sequences.\n",
      "Sample 158 - 17970 reads in 15933 unique sequences.\n",
      "Sample 159 - 19081 reads in 14759 unique sequences.\n",
      "Sample 160 - 13009 reads in 10478 unique sequences.\n",
      "Sample 161 - 76 reads in 75 unique sequences.\n",
      "Sample 162 - 15740 reads in 14027 unique sequences.\n",
      "Sample 163 - 14533 reads in 12772 unique sequences.\n",
      "Sample 164 - 10911 reads in 9969 unique sequences.\n",
      "Sample 165 - 17851 reads in 15803 unique sequences.\n",
      "Sample 166 - 20832 reads in 16989 unique sequences.\n",
      "Sample 167 - 16072 reads in 13265 unique sequences.\n",
      "Sample 168 - 16669 reads in 14839 unique sequences.\n",
      "Sample 169 - 24933 reads in 21608 unique sequences.\n",
      "Sample 170 - 7154 reads in 6454 unique sequences.\n",
      "Sample 171 - 8909 reads in 7960 unique sequences.\n",
      "Sample 172 - 12455 reads in 9703 unique sequences.\n",
      "Sample 173 - 15163 reads in 11595 unique sequences.\n",
      "Sample 174 - 11682 reads in 10220 unique sequences.\n",
      "Sample 175 - 12154 reads in 10752 unique sequences.\n",
      "Sample 176 - 10886 reads in 9660 unique sequences.\n",
      "Sample 177 - 13619 reads in 12203 unique sequences.\n",
      "Sample 178 - 20200 reads in 18250 unique sequences.\n",
      "Sample 179 - 17006 reads in 15496 unique sequences.\n",
      "Sample 180 - 14727 reads in 13290 unique sequences.\n",
      "Sample 181 - 219 reads in 216 unique sequences.\n",
      "Sample 182 - 11194 reads in 10454 unique sequences.\n",
      "Sample 183 - 10665 reads in 9851 unique sequences.\n",
      "Sample 184 - 9193 reads in 8095 unique sequences.\n",
      "Sample 185 - 19886 reads in 16959 unique sequences.\n",
      "Sample 186 - 8968 reads in 8143 unique sequences.\n",
      "Sample 187 - 9094 reads in 8250 unique sequences.\n",
      "Sample 188 - 12919 reads in 11848 unique sequences.\n",
      "Sample 189 - 14933 reads in 13123 unique sequences.\n",
      "Sample 190 - 16075 reads in 14222 unique sequences.\n",
      "Sample 191 - 9022 reads in 7789 unique sequences.\n",
      "Sample 192 - 8924 reads in 7636 unique sequences.\n",
      "Sample 193 - 36 reads in 35 unique sequences.\n",
      "Sample 194 - 5777 reads in 5366 unique sequences.\n",
      "Sample 195 - 6468 reads in 5952 unique sequences.\n",
      "Sample 196 - 5824 reads in 5524 unique sequences.\n",
      "Sample 197 - 22203 reads in 20672 unique sequences.\n",
      "Sample 198 - 19107 reads in 17334 unique sequences.\n",
      "Sample 199 - 15149 reads in 13276 unique sequences.\n",
      "Sample 200 - 22163 reads in 17397 unique sequences.\n",
      "Sample 201 - 5708 reads in 4799 unique sequences.\n",
      "Sample 202 - 15428 reads in 13712 unique sequences.\n",
      "Sample 203 - 9472 reads in 8618 unique sequences.\n",
      "Sample 204 - 11634 reads in 9125 unique sequences.\n",
      "Sample 205 - 8999 reads in 6690 unique sequences.\n",
      "Sample 206 - 9589 reads in 8638 unique sequences.\n",
      "Sample 207 - 13685 reads in 12204 unique sequences.\n",
      "Sample 208 - 7164 reads in 6387 unique sequences.\n",
      "Sample 209 - 9816 reads in 8234 unique sequences.\n",
      "Sample 210 - 6168 reads in 5630 unique sequences.\n",
      "Sample 211 - 11922 reads in 10980 unique sequences.\n",
      "Sample 212 - 29240 reads in 22926 unique sequences.\n",
      "Sample 213 - 836 reads in 741 unique sequences.\n",
      "Sample 214 - 6156 reads in 5709 unique sequences.\n",
      "Sample 215 - 12434 reads in 11199 unique sequences.\n",
      "Sample 216 - 32313 reads in 28401 unique sequences.\n",
      "Sample 217 - 9849 reads in 9133 unique sequences.\n",
      "Sample 218 - 9397 reads in 8500 unique sequences.\n",
      "   selfConsist step 2 \n",
      "   selfConsist step 3 \n",
      "   selfConsist step 4 \n",
      "   selfConsist step 5 \n",
      "\n",
      "\n",
      "Convergence after  5  rounds.\n",
      "2b) Reverse Reads\n",
      "Initializing error rates to maximum possible estimate.\n",
      "Sample 1 - 23019 reads in 16784 unique sequences.\n",
      "Sample 2 - 8541 reads in 7000 unique sequences.\n",
      "Sample 3 - 17795 reads in 12927 unique sequences.\n",
      "Sample 4 - 10717 reads in 8501 unique sequences.\n",
      "Sample 5 - 18306 reads in 14424 unique sequences.\n",
      "Sample 6 - 11676 reads in 8412 unique sequences.\n",
      "Sample 7 - 9082 reads in 6496 unique sequences.\n",
      "Sample 8 - 9944 reads in 6446 unique sequences.\n",
      "Sample 9 - 18467 reads in 13868 unique sequences.\n",
      "Sample 10 - 11020 reads in 8741 unique sequences.\n",
      "Sample 11 - 162 reads in 151 unique sequences.\n",
      "Sample 12 - 9990 reads in 7009 unique sequences.\n",
      "Sample 13 - 12309 reads in 8264 unique sequences.\n",
      "Sample 14 - 17294 reads in 13339 unique sequences.\n",
      "Sample 15 - 15253 reads in 11957 unique sequences.\n",
      "Sample 16 - 13013 reads in 8181 unique sequences.\n",
      "Sample 17 - 16632 reads in 10728 unique sequences.\n",
      "Sample 18 - 12699 reads in 9066 unique sequences.\n",
      "Sample 19 - 14950 reads in 11403 unique sequences.\n",
      "Sample 20 - 17943 reads in 12985 unique sequences.\n",
      "Sample 21 - 15274 reads in 11338 unique sequences.\n",
      "Sample 22 - 8519 reads in 7505 unique sequences.\n",
      "Sample 23 - 20508 reads in 17539 unique sequences.\n",
      "Sample 24 - 14753 reads in 9868 unique sequences.\n",
      "Sample 25 - 14557 reads in 9805 unique sequences.\n",
      "Sample 26 - 12832 reads in 9470 unique sequences.\n",
      "Sample 27 - 6172 reads in 4858 unique sequences.\n",
      "Sample 28 - 10076 reads in 8322 unique sequences.\n",
      "Sample 29 - 15731 reads in 13024 unique sequences.\n",
      "Sample 30 - 14988 reads in 11961 unique sequences.\n",
      "Sample 31 - 17632 reads in 13737 unique sequences.\n",
      "Sample 32 - 9583 reads in 7974 unique sequences.\n",
      "Sample 33 - 33268 reads in 21648 unique sequences.\n",
      "Sample 34 - 16062 reads in 11697 unique sequences.\n",
      "Sample 35 - 2 reads in 2 unique sequences.\n",
      "Sample 36 - 17794 reads in 11547 unique sequences.\n",
      "Sample 37 - 23211 reads in 15142 unique sequences.\n",
      "Sample 38 - 14487 reads in 10564 unique sequences.\n",
      "Sample 39 - 18197 reads in 13085 unique sequences.\n",
      "Sample 40 - 11899 reads in 10305 unique sequences.\n",
      "Sample 41 - 10514 reads in 9115 unique sequences.\n",
      "Sample 42 - 30090 reads in 18185 unique sequences.\n",
      "Sample 43 - 25908 reads in 15422 unique sequences.\n",
      "Sample 44 - 10997 reads in 7872 unique sequences.\n",
      "Sample 45 - 8958 reads in 6910 unique sequences.\n",
      "Sample 46 - 9653 reads in 6656 unique sequences.\n",
      "Sample 47 - 17169 reads in 11535 unique sequences.\n",
      "Sample 48 - 19525 reads in 12701 unique sequences.\n",
      "Sample 49 - 10079 reads in 6632 unique sequences.\n",
      "Sample 50 - 15964 reads in 12318 unique sequences.\n",
      "Sample 51 - 12976 reads in 9997 unique sequences.\n",
      "Sample 52 - 21186 reads in 17645 unique sequences.\n",
      "Sample 53 - 19489 reads in 15887 unique sequences.\n",
      "Sample 54 - 20982 reads in 15771 unique sequences.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample 55 - 15116 reads in 12686 unique sequences.\n",
      "Sample 56 - 15638 reads in 12494 unique sequences.\n",
      "Sample 57 - 8881 reads in 6897 unique sequences.\n",
      "Sample 58 - 16211 reads in 11768 unique sequences.\n",
      "Sample 59 - 7 reads in 7 unique sequences.\n",
      "Sample 60 - 3634 reads in 2987 unique sequences.\n",
      "Sample 61 - 13650 reads in 10297 unique sequences.\n",
      "Sample 62 - 16223 reads in 12218 unique sequences.\n",
      "Sample 63 - 19122 reads in 13493 unique sequences.\n",
      "Sample 64 - 14891 reads in 11278 unique sequences.\n",
      "Sample 65 - 18187 reads in 13644 unique sequences.\n",
      "Sample 66 - 14281 reads in 11467 unique sequences.\n",
      "Sample 67 - 18784 reads in 14587 unique sequences.\n",
      "Sample 68 - 9472 reads in 8490 unique sequences.\n",
      "Sample 69 - 9864 reads in 8729 unique sequences.\n",
      "Sample 70 - 27853 reads in 21358 unique sequences.\n",
      "Sample 71 - 5381 reads in 4711 unique sequences.\n",
      "Sample 72 - 9324 reads in 7773 unique sequences.\n",
      "Sample 73 - 6624 reads in 5728 unique sequences.\n",
      "Sample 74 - 9045 reads in 6846 unique sequences.\n",
      "Sample 75 - 14406 reads in 10264 unique sequences.\n",
      "Sample 76 - 11569 reads in 9457 unique sequences.\n",
      "Sample 77 - 9786 reads in 8088 unique sequences.\n",
      "Sample 78 - 27061 reads in 18150 unique sequences.\n",
      "Sample 79 - 15770 reads in 11247 unique sequences.\n",
      "Sample 80 - 8457 reads in 6219 unique sequences.\n",
      "Sample 81 - 11 reads in 11 unique sequences.\n",
      "Sample 82 - 13 reads in 13 unique sequences.\n",
      "Sample 83 - 8926 reads in 7013 unique sequences.\n",
      "Sample 84 - 6880 reads in 5595 unique sequences.\n",
      "Sample 85 - 7 reads in 7 unique sequences.\n",
      "Sample 86 - 4 reads in 4 unique sequences.\n",
      "Sample 87 - 2 reads in 2 unique sequences.\n",
      "Sample 88 - 5 reads in 5 unique sequences.\n",
      "Sample 89 - 15939 reads in 12127 unique sequences.\n",
      "Sample 90 - 9472 reads in 7930 unique sequences.\n",
      "Sample 91 - 10720 reads in 8303 unique sequences.\n",
      "Sample 92 - 18560 reads in 14113 unique sequences.\n",
      "Sample 93 - 10087 reads in 7689 unique sequences.\n",
      "Sample 94 - 10315 reads in 7914 unique sequences.\n",
      "Sample 95 - 7 reads in 7 unique sequences.\n",
      "Sample 96 - 7283 reads in 6458 unique sequences.\n",
      "Sample 97 - 12678 reads in 10953 unique sequences.\n",
      "Sample 98 - 21797 reads in 17914 unique sequences.\n",
      "Sample 99 - 9818 reads in 8211 unique sequences.\n",
      "Sample 100 - 18792 reads in 15295 unique sequences.\n",
      "Sample 101 - 40141 reads in 29478 unique sequences.\n",
      "Sample 102 - 36355 reads in 26250 unique sequences.\n",
      "Sample 103 - 613 reads in 587 unique sequences.\n",
      "Sample 104 - 6690 reads in 5997 unique sequences.\n",
      "Sample 105 - 9309 reads in 8091 unique sequences.\n",
      "Sample 106 - 1305 reads in 1078 unique sequences.\n",
      "Sample 107 - 945 reads in 783 unique sequences.\n",
      "Sample 108 - 6021 reads in 4541 unique sequences.\n",
      "Sample 109 - 6794 reads in 6006 unique sequences.\n",
      "Sample 110 - 5219 reads in 4640 unique sequences.\n",
      "Sample 111 - 12329 reads in 10079 unique sequences.\n",
      "Sample 112 - 17092 reads in 13424 unique sequences.\n",
      "Sample 113 - 68 reads in 67 unique sequences.\n",
      "Sample 114 - 16131 reads in 13861 unique sequences.\n",
      "Sample 115 - 12525 reads in 10784 unique sequences.\n",
      "Sample 116 - 12921 reads in 10822 unique sequences.\n",
      "Sample 117 - 12197 reads in 10373 unique sequences.\n",
      "Sample 118 - 9762 reads in 8155 unique sequences.\n",
      "Sample 119 - 12402 reads in 10193 unique sequences.\n",
      "Sample 120 - 11245 reads in 9330 unique sequences.\n",
      "Sample 121 - 4807 reads in 4223 unique sequences.\n",
      "Sample 122 - 642 reads in 595 unique sequences.\n",
      "Sample 123 - 15606 reads in 13180 unique sequences.\n",
      "Sample 124 - 15298 reads in 12673 unique sequences.\n",
      "Sample 125 - 12000 reads in 10080 unique sequences.\n",
      "Sample 126 - 9152 reads in 7606 unique sequences.\n",
      "Sample 127 - 8130 reads in 6887 unique sequences.\n",
      "Sample 128 - 11297 reads in 8080 unique sequences.\n",
      "Sample 129 - 14584 reads in 10666 unique sequences.\n",
      "Sample 130 - 9790 reads in 7776 unique sequences.\n",
      "Sample 131 - 14328 reads in 10273 unique sequences.\n",
      "Sample 132 - 27570 reads in 21613 unique sequences.\n",
      "Sample 133 - 13890 reads in 9855 unique sequences.\n",
      "Sample 134 - 21011 reads in 12225 unique sequences.\n",
      "Sample 135 - 25225 reads in 20082 unique sequences.\n",
      "Sample 136 - 26349 reads in 16554 unique sequences.\n",
      "Sample 137 - 30769 reads in 19106 unique sequences.\n",
      "Sample 138 - 23586 reads in 16930 unique sequences.\n",
      "Sample 139 - 14470 reads in 11870 unique sequences.\n",
      "Sample 140 - 16157 reads in 11940 unique sequences.\n",
      "Sample 141 - 46411 reads in 30852 unique sequences.\n",
      "Sample 142 - 26002 reads in 20569 unique sequences.\n",
      "Sample 143 - 15312 reads in 13017 unique sequences.\n",
      "Sample 144 - 23646 reads in 18540 unique sequences.\n",
      "Sample 145 - 14398 reads in 11293 unique sequences.\n",
      "Sample 146 - 14621 reads in 11715 unique sequences.\n",
      "Sample 147 - 20055 reads in 15662 unique sequences.\n",
      "Sample 148 - 15231 reads in 13192 unique sequences.\n",
      "Sample 149 - 13383 reads in 11672 unique sequences.\n",
      "Sample 150 - 16846 reads in 12786 unique sequences.\n",
      "Sample 151 - 12330 reads in 9611 unique sequences.\n",
      "Sample 152 - 11638 reads in 9159 unique sequences.\n",
      "Sample 153 - 13862 reads in 10543 unique sequences.\n",
      "Sample 154 - 14304 reads in 10562 unique sequences.\n",
      "Sample 155 - 16302 reads in 12572 unique sequences.\n",
      "Sample 156 - 10116 reads in 8245 unique sequences.\n",
      "Sample 157 - 12 reads in 12 unique sequences.\n",
      "Sample 158 - 17970 reads in 12737 unique sequences.\n",
      "Sample 159 - 19081 reads in 14264 unique sequences.\n",
      "Sample 160 - 13009 reads in 9626 unique sequences.\n",
      "Sample 161 - 76 reads in 67 unique sequences.\n",
      "Sample 162 - 15740 reads in 12631 unique sequences.\n",
      "Sample 163 - 14533 reads in 11858 unique sequences.\n",
      "Sample 164 - 10911 reads in 8987 unique sequences.\n",
      "Sample 165 - 17851 reads in 14412 unique sequences.\n",
      "Sample 166 - 20832 reads in 17151 unique sequences.\n",
      "Sample 167 - 16072 reads in 13204 unique sequences.\n",
      "Sample 168 - 16669 reads in 13680 unique sequences.\n",
      "Sample 169 - 24933 reads in 19723 unique sequences.\n",
      "Sample 170 - 7154 reads in 5759 unique sequences.\n",
      "Sample 171 - 8909 reads in 6962 unique sequences.\n",
      "Sample 172 - 12455 reads in 9733 unique sequences.\n",
      "Sample 173 - 15163 reads in 11509 unique sequences.\n",
      "Sample 174 - 11682 reads in 8784 unique sequences.\n",
      "Sample 175 - 12154 reads in 9741 unique sequences.\n",
      "Sample 176 - 10886 reads in 8487 unique sequences.\n",
      "Sample 177 - 13619 reads in 11070 unique sequences.\n",
      "Sample 178 - 20200 reads in 13025 unique sequences.\n",
      "Sample 179 - 17006 reads in 13544 unique sequences.\n",
      "Sample 180 - 14727 reads in 12202 unique sequences.\n",
      "Sample 181 - 219 reads in 202 unique sequences.\n",
      "Sample 182 - 11194 reads in 8699 unique sequences.\n",
      "Sample 183 - 10665 reads in 8429 unique sequences.\n",
      "Sample 184 - 9193 reads in 7685 unique sequences.\n",
      "Sample 185 - 19886 reads in 15686 unique sequences.\n",
      "Sample 186 - 8968 reads in 6764 unique sequences.\n",
      "Sample 187 - 9094 reads in 6679 unique sequences.\n",
      "Sample 188 - 12919 reads in 10976 unique sequences.\n",
      "Sample 189 - 14933 reads in 12251 unique sequences.\n",
      "Sample 190 - 16075 reads in 12229 unique sequences.\n",
      "Sample 191 - 9022 reads in 7612 unique sequences.\n",
      "Sample 192 - 8924 reads in 7486 unique sequences.\n",
      "Sample 193 - 36 reads in 36 unique sequences.\n",
      "Sample 194 - 5777 reads in 5309 unique sequences.\n",
      "Sample 195 - 6468 reads in 5909 unique sequences.\n",
      "Sample 196 - 5824 reads in 5069 unique sequences.\n",
      "Sample 197 - 22203 reads in 17737 unique sequences.\n",
      "Sample 198 - 19107 reads in 16486 unique sequences.\n",
      "Sample 199 - 15149 reads in 12633 unique sequences.\n",
      "Sample 200 - 22163 reads in 17529 unique sequences.\n",
      "Sample 201 - 5708 reads in 4920 unique sequences.\n",
      "Sample 202 - 15428 reads in 12387 unique sequences.\n",
      "Sample 203 - 9472 reads in 8302 unique sequences.\n",
      "Sample 204 - 11634 reads in 8664 unique sequences.\n",
      "Sample 205 - 8999 reads in 6729 unique sequences.\n",
      "Sample 206 - 9589 reads in 7875 unique sequences.\n",
      "Sample 207 - 13685 reads in 11056 unique sequences.\n",
      "Sample 208 - 7164 reads in 6302 unique sequences.\n",
      "Sample 209 - 9816 reads in 7352 unique sequences.\n",
      "Sample 210 - 6168 reads in 5276 unique sequences.\n",
      "Sample 211 - 11922 reads in 8928 unique sequences.\n",
      "Sample 212 - 29240 reads in 22528 unique sequences.\n",
      "Sample 213 - 836 reads in 731 unique sequences.\n",
      "Sample 214 - 6156 reads in 5549 unique sequences.\n",
      "Sample 215 - 12434 reads in 9885 unique sequences.\n",
      "Sample 216 - 32313 reads in 21122 unique sequences.\n",
      "Sample 217 - 9849 reads in 6561 unique sequences.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample 218 - 9397 reads in 7724 unique sequences.\n",
      "   selfConsist step 2 \n",
      "   selfConsist step 3 \n",
      "   selfConsist step 4 \n",
      "   selfConsist step 5 \n",
      "   selfConsist step 6 \n",
      "   selfConsist step 7 \n",
      "\n",
      "\n",
      "Convergence after  7  rounds.\n",
      "\n",
      "Duplicate sequences in merged output.\n",
      "Duplicate sequences in merged output.\n",
      "Duplicate sequences in merged output.\n",
      "3) Denoise remaining samples \n",
      "Duplicate sequences detected and merged.\n",
      "Duplicate sequences detected and merged.\n",
      "Duplicate sequences detected and merged.\n",
      "The sequences being tabled vary in length.\n",
      "4) Remove chimeras (method = consensus)\n",
      "5) Write output\n",
      "\u001b[32mSaved FeatureTable[Frequency] to: table.qza\u001b[0m\n",
      "\u001b[32mSaved FeatureData[Sequence] to: rep-seqs.qza\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "# Initially, using trim left and right at 300 resulted in throwing out a lot of sequences.\n",
    "# Because the 300bp paired ends sequencing has... issues, this meant that a lot of samples dropped out before 300,\n",
    "# And so were thrown out. Hence the initial results with very few samples passing through.\n",
    "\n",
    "# Had weird low results the second time around with trunc-len of 100 \n",
    "# That's because lots of samples were getting through but they weren't merging because I cut them off at 100.\n",
    "\n",
    "# Then, thought 298 and 275 cutoffs could be a good in between value. \n",
    "# Could play with these values to see effect on end data.\n",
    "# Taylor et al. (2016) paper for these primers suggested amplicon should be between 267-511 bp.\n",
    "# We want 20bp overlap for alignment\n",
    "# So, if we have 286 fwd read and then 230 rev read\n",
    "# That leaves us 286 -20 +230 -20 +20 = 286+230-20=286+210 = 496bp potential length, so, a little short.\n",
    "# i.e., the samples wouldn't align properly for the longest amplicons.\n",
    "# The question is, is lengthening the trimmed reads going to result in more issues?\n",
    "# Expect to need a total of ~5bp more total - could go to 291 or 235...\n",
    "\n",
    "# Maybe the issue was that some amplicons were ONLY 267bp... so of course the \"error\" rates go off the charts-\n",
    "# There is no amplicon to be sequenced. That would make sense... hence, using 0 for truncation lengths.\n",
    "# That was correct - need to use a length of 0 (no cutoffs), which means some samples have very high error rates at the end\n",
    "# So, use ~high maxee (recommended by Dada2 author in online help)\n",
    "\n",
    "!qiime dada2 denoise-paired --i-demultiplexed-seqs demux.qza --p-n-reads-learn 3000000 --p-trim-left-f 0 --p-trim-left-r 0 --p-trunc-len-f 0 --p-trunc-len-r 0 --p-max-ee 5 --p-n-threads 0 --verbose --o-table table --o-representative-sequences rep-seqs\n",
    "\n",
    "# Increasing the maxee maxes out at a certain point.\n",
    "# Using p-n-reads-learn 3M to look at all reads - 3x longer, probably, but may have high heterogeneity in system\n",
    "# Should re-check 16S data for this potential issue.\n",
    "\n",
    "# On my computer, for full run data,\n",
    "# Started 11:59 PM\n",
    "# Up to self-consist step 3 was running on the second reads by 6:30AM; Step 4 at 7:16AM; Step 5 at 8:20AM\n",
    "# That's about 1.2 runs per hour - so maybe 2.4h left if up to 5 self-consist on the last one.\n",
    "# Remember, however, I made it look at all the sequences, not just a 1M subset."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[32mSaved Visualization to: table.qzv\u001b[0m\n",
      "\u001b[32mSaved Visualization to: rep-seqs.qzv\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "!qiime feature-table summarize --i-table table.qza --o-visualization table.qzv\n",
    "\n",
    "!qiime feature-table tabulate-seqs --i-data rep-seqs.qza --o-visualization rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting.\n",
      "Press the 'q' key, Control-C, or Control-D to quit. This view may no longer be accessible or work correctly after quitting."
     ]
    }
   ],
   "source": [
    "!qiime tools view rep-seqs.qzv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Output looks pretty good - next steps would be to assign correct sample IDs from barcodes\n",
    "# and to use ITSx to make sure we only have the ITS2 sequences, with less-informative end sequences removed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Save it all in a format we can use in R:\n",
    "!mkdir OTU_table\n",
    "\n",
    "!qiime tools export table.qza --output-dir OTU_table\n",
    "!qiime tools export rep-seqs.qza --output-dir OTU_table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# Assigning taxonomy using ITS2 regions\n",
    "# Ran in shell, not here.\n",
    "# So, some of the OTUs should have no match...\n",
    "#!parallel_assign_taxonomy_uclust.py -i OTU_table/dna-sequences-ITSx.fasta.ITS2.fasta -o OTU_table/taxonomy -r ../data/db/sh_refs_qiime_ver7_99_28.06.2017.fasta -t ../data/db/sh_taxonomy_qiime_ver7_99_28.06.2017.txt -O 4"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Usage: biom subset-table [OPTIONS]\r\n",
      "\r\n",
      "  Subset a BIOM table.\r\n",
      "\r\n",
      "  Subset a BIOM table, over either observations or samples, without fully\r\n",
      "  parsing it. This command is intended to assist in working with very large\r\n",
      "  tables when tight on memory, or as a lightweight way to subset a full\r\n",
      "  table. Currently, it is possible to produce tables with rows or columns\r\n",
      "  (observations or samples) that are fully zeroed.\r\n",
      "\r\n",
      "  Example usage:\r\n",
      "\r\n",
      "  Choose a subset of the observations in table.biom (JSON) and write them to\r\n",
      "  subset.biom:\r\n",
      "\r\n",
      "  $ biom subset-table -j table.biom -a observations -s observation_ids.txt\r\n",
      "  -o subset.biom\r\n",
      "\r\n",
      "  Choose a subset of the observations in table.biom (HDF5) and write them to\r\n",
      "  subset.biom:\r\n",
      "\r\n",
      "  $ biom subset-table -i table.biom -a observations -s observation_ids.txt\r\n",
      "  -o subset.biom\r\n",
      "\r\n",
      "Options:\r\n",
      "  -i, --input-hdf5-fp PATH        the input hdf5 BIOM table filepath to subset\r\n",
      "  -j, --input-json-fp PATH        the input json BIOM table filepath to subset\r\n",
      "  -a, --axis [sample|observation]\r\n",
      "                                  the axis to subset over, either sample or\r\n",
      "                                  observation  [required]\r\n",
      "  -s, --ids PATH                  a file containing a single column of IDs to\r\n",
      "                                  retain (either sample IDs or observation\r\n",
      "                                  IDs, depending on the axis)  [required]\r\n",
      "  -o, --output-fp PATH            the output BIOM table filepath  [required]\r\n",
      "  --help                          Show this message and exit.\r\n"
     ]
    }
   ],
   "source": [
    "!biom subset-table --help"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!biom add-metadata -i OTU_table/feature-table.biom -o OTU_table/feature-table.metaD.biom --sample-metadata-fp ../data/Soils_data/WBNPNWT_Soils_2015_Metadata_File_ITS.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!grep \">\" OTU_table/dna-sequences-ITSx.fasta.ITS2.fasta > OTU_table/dna-sequences-ITSx.fasta.ITS2.names.txt\n",
    "# Takes all the sequence names that made it through the ITSx filter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "! cat OTU_table/dna-sequences-ITSx.fasta.ITS2.names.txt | cut -f2- -d\\> > OTU_table/ITS2.names.txt\n",
    "# Cuts out the \">\" in their name so we can feed them into subset-table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9d21cc18ee8717b08f3eb5e788f93f28\r\n",
      "811b6adce92e17ac67df13b027edf772\r\n",
      "c19acfee4940ac92595298d3b15e50c6\r\n",
      "04cd8bfe2f9bf8f3b4b16fe01f1b0c59\r\n",
      "6dcbf1fcf1eec33ef98e29ab28a830fc\r\n",
      "14bfa01fdc1730d27d4ec723508ca082\r\n",
      "d647db11e143d8e73e3ccd87a7d5d6ca\r\n",
      "844c40fec6a1660425606681bed19230\r\n",
      "6bd11a001a9543f1bd22796d5e089717\r\n",
      "b50cfae57c5213a323d53ac76f058fea\r\n"
     ]
    }
   ],
   "source": [
    "!head -10 OTU_table/ITS2.names.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!biom subset-table -i OTU_table/feature-table.metaD.biom -o OTU_table/feature-table.metaD.subset.biom -a observation -s OTU_table/ITS2.names.txt "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "# That does remove the ~20 samples that didn't match an ITS."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!biom add-metadata -i OTU_table/feature-table.metaD.subset.biom -o OTU_table/feature-table.metaD.subset.tax.biom --observation-metadata-fp OTU_table/taxonomy/dna-sequences-ITSx.fasta.ITS2_tax_assignments.txt --sc-separated taxonomy --observation-header OTUID,taxonomy\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!biom summarize-table -i OTU_table/feature-table.metaD.subset.tax.biom -o OTU_table/feature-table.metaD.subset.tax.biom.summary.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Num samples: 206\r\n",
      "Num observations: 1031\r\n",
      "Total count: 2305144\r\n",
      "Table density (fraction of non-zero values): 0.018\r\n",
      "\r\n",
      "Counts/sample summary:\r\n",
      " Min: 76.0\r\n",
      " Max: 38397.0\r\n",
      " Median: 10552.000\r\n",
      " Mean: 11190.019\r\n",
      " Std. dev.: 6027.102\r\n",
      " Sample Metadata Categories: bui; CFSI; Exch_Ca_mg_kg; O_Depth_cm; Al_mg_kg; Na_mg_kg; fwi; P_mg_kg; Clay_pct; pH; Zn_mg_kg; Exch_Na_mg_kg; TOC_LOI_pct; Mo_mg_kg; ws; K_mg_kg; CEC_cmol_kg; Sample_Name; Fire_ID; Moisture; Revcomp_Rev_Primer_Barcode; Total_N_pct; Plains; Exch_K_mg_kg; Nutrient; Dead_Trees; Burned_Unburned; CBI; Land_Class; Pct_Exposed_Mineral; Site_ID; Rev_Primer_Barcode; Ca_mg_kg; Live_Trees; rh; prec; ffmc; Burn_Severity_Index; RBR; Sample_ID; Mn_mg_kg; Fe_mg_kg; Project_ID; Severity_Class; Barcodes; temp; Veg_Comm; Mean_Duff_Depth_cm; S_mg_kg; Moisture_Regime; Mg_mg_kg; TOC_HCL_cruc_pct; Overstory_CBI; isi; TC_pct; Understory_CBI; Fwd_Primer_Barcode; Sand_pct; Correct_ID; Cu_mg_kg; Interval; dmc; Forest; EC_mS_cm; Silt_pct; Org_or_Min; Community; Land_Class_Unburned; nTrees; dc; TIC_ash_pct; Exch_Mg_mg_kg; Total_S_pct; Ecosite; Replicate\r\n",
      " Observation Metadata Categories: taxonomy\r\n",
      "\r\n",
      "Counts/sample detail:\r\n",
      "15S-WB-10O-1: 76.0\r\n",
      "15S-NT-22M-1: 145.0\r\n",
      "15S-WB-16M-1: 193.0\r\n",
      "15S-NT-50O-1: 200.0\r\n",
      "15S-NT-U09O-2: 397.0\r\n",
      "Blank-16: 689.0\r\n",
      "15S-NT-52O-1: 762.0\r\n",
      "15S-NT-51O-2: 1172.0\r\n",
      "15S-WB-U01O-2: 2186.0\r\n",
      "15S-NT-36O-2: 2592.0\r\n",
      "15S-NT-40O-1: 2595.0\r\n",
      "15S-WB-12O-2: 2846.0\r\n",
      "15S-NT-U09O-1: 2972.0\r\n",
      "15S-WB-U02O-1: 3059.0\r\n",
      "15S-NT-U06O-2: 3060.0\r\n",
      "Blank-17: 3477.0\r\n",
      "15S-WB-U02O-2: 3678.0\r\n",
      "15S-NT-50O-2: 4108.0\r\n",
      "15S-NT-U06O-1: 4300.0\r\n",
      "Blank-10: 4315.0\r\n",
      "15S-WB-U04M-1: 4345.0\r\n",
      "15S-WB-06O-2: 4398.0\r\n",
      "15S-NT-52O-2: 4501.0\r\n",
      "15S-NT-47O-2: 4881.0\r\n",
      "15S-NT-41M-1: 4915.0\r\n"
     ]
    }
   ],
   "source": [
    "!head -40 OTU_table/feature-table.metaD.subset.tax.biom.summary.txt\n",
    "# Beware! Sequence names are misassigned by the sequencing centre.\n",
    "# Need to rename - correct sample ID currently stored as Correct_ID\n",
    "\n",
    "# There were 2896855 sequences that went in. So, 80% of sequences were maintained and matched some OTU.\n",
    "# Not awful, not amazing.\n",
    "# If more seqs were maintained, that would likely add more OTUs to the system... which would populate the low-abundance\n",
    "# sequences... and it might actually pick up more sequences across samples\n",
    "# Because the POOL option in dada2 allows it to compare across samples\n",
    "# And we have O and M horizon site \"replicates\", and lab-DNA extraction replicates, so that really might help pull\n",
    "# out more real OTUs."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Final files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 258,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "!cp OTU_table/feature-table.metaD.subset.tax.biom OTU_table/ITS-OTU-table.biom\n",
    "!cp OTU_table/dna-sequences-ITSx.fasta.ITS2.fasta OTU_table/OTU-seqs.fasta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Final files are ITS-OTU-table.biom and OTU-seqs.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
